packages/teleterm/src/mainProcess/clusterContextMenu.ts

```diff
@@ -0,0 +1,107 @@
+import { ipcMain, ipcRenderer, Menu } from 'electron';
+
+enum ClusterContextMenuEventChannel {
+  ToMain = 'main-process-cluster-context-menu-to-main',
+  ToRenderer = 'main-process-cluster-context-menu-to-renderer',
+}
+
+enum ClusterContextMenuEventType {
+  Refresh = 'Refresh',
+  Login = 'Login',
+  Logout = 'Logout',
+  Remove = 'Remove',
+}
+
+export interface ClusterContextMenuOptions {
+  isClusterConnected: boolean;
+
+  onRefresh(): void;
+
+  onLogin(): void;
+
+  onLogout(): void;
+
+  onRemove(): void;
+}
+
+type MainClusterContextMenuOptions = Pick<
+  ClusterContextMenuOptions,
+  'isClusterConnected'
+>;
+
+export function subscribeToClusterContextMenuEvent(): void {
+  ipcMain.on(
+    ClusterContextMenuEventChannel.ToMain,
```

Comment:
Do we need separate message types per Main vs Render?

Reply:
Yes, I was basing on the official documentation, where they use separate channels for bidirectional communication:  https://www.electronjs.org/docs/latest/api/ipc-main 
```typescript
ipcMain.on('asynchronous-message', (event, arg) => {
  console.log(arg) // prints "ping"
  event.reply('asynchronous-reply', 'pong')
})
```

But another idea came to my mind - communication with promises & [ipc.handle](https://www.electronjs.org/docs/latest/api/ipc-main#ipcmainhandlechannel-listener)
I think I like it, it's a bit shorter and simpler
https://gist.github.com/gzdunek/fe3333102278e0738355afd7df389d98